packages/application-shell/src/components/application-shell/application-shell.js

```diff
@@ -97,241 +96,253 @@ export const RestrictedApplication = props => (
 
-                                // when used outside of a project context,
-                                // or when the project is expired or supsended
-                                const useProjectContext =
-                                  project &&
-                                  !(
-                                    project.suspension?.isActive ||
-                                    project.expiry?.isActive
-                                  );
+                                  // when used outside of a project context,
+                                  // or when the project is expired or supsended
+                                  const useProjectContext =
+                                    project &&
+                                    !(
+                                      project.suspension?.isActive ||
+                                      project.expiry?.isActive
+                                    );
 
-                                if (!useProjectContext)
-                                  return (
-                                    <QuickAccess
-                                      history={routeProps.history}
-                                      user={user}
-                                    />
-                                  );
-                                return (
-                                  <ProjectDataLocale
-                                    locales={project.languages}
-                                  >
-                                    {({ locale, setProjectDataLocale }) => (
+                                  if (!useProjectContext)
+                                    return (
                                       <QuickAccess
-                                        project={project}
-                                        projectDataLocale={locale}
-                                        onChangeProjectDataLocale={
-                                          setProjectDataLocale
-                                        }
                                         history={routeProps.history}
                                         user={user}
                                       />
-                                    )}
-                                  </ProjectDataLocale>
+                                    );
+                                  return (
+                                    <ProjectDataLocale
+                                      locales={project.languages}
+                                    >
+                                      {({ locale, setProjectDataLocale }) => (
+                                        <ApplicationStateProvider
+                                          user={user}
+                                          project={project}
+                                          projectDataLocale={locale}
+                                          environment={props.environment}
+                                        >
```

Comment:
At this point, we know also the `projectKey`, so we put all the info in the provider again